Add TypeScript Guides (#6454)
DevExpress/devextreme-documentation: concepts/Common/TypeScript Guides/05 Scopes of Types/20 Internal Types.md

> Internal types are for internal implementation only. **Avoid using them in your projects.** They have a special warning:

Reviewer: Removing the repetition of "internal" and avoiding the vague/weak verb "have". 
```suggestion
Internal types support DevExtreme infrastructure and are not a part of component API. **Avoid using these types in your projects.** If you do, they issue the following warning:
```

> Internal types support DevExtreme infrastructure and are not a part of component API. **Avoid using these types in your projects.** If you do, they issue the following warning:

Reviewer 2: ```suggestion
Internal types support DevExtreme infrastructure and are not a part of the component's API. **Avoid using these types in your projects.** If you do, they issue the following warning:
```

    Attention! This type is for internal purposes only. If you used it previously, please submit a ticket to our Support Center. We will check if there is an alternative solution.

> You will see this warning when navigating to the type declaration, in the console during build, or in your IDE, depending on its settings.

Reviewer: I wanted to change the last comma to parentheses, because otherwise it seems like we're listing yet another place where you can see the message. 

```suggestion
You will see this warning when navigating to the type declaration, in the console during build, or in your IDE (depending on its settings).
```

> You will see this warning when navigating to the type declaration, in the console during build, or in your IDE (depending on its settings).

Reviewer 2: ```suggestion
This warning is shown when navigating to the type declaration, in the console during build, or in your IDE (depending on its settings).
```
